lessons/06-tool-calling/README.md

```diff
@@ -0,0 +1,426 @@
+
+<div>
+    <img src="./assets/amelia.jpeg" alt="Amelia piloting a plane" width="300">
+</div>
+
+**You**: "Amelia, is that you?"
+
+**Amelia Earhart**: "Who are you? Let me guess, Ada sent you, right?"
+
+**You**: "Yes, correct. I'm here to help you, I think. Ada wasn't big on specifics."
+
+**Amelia Earhart**: "Well, good thing you're here. I'm in a bit of a pickle. I'm trying to find a place to land, and I'm running out of fuel. Can you help me?"
+
+**You**: "Device, can you tell me more about Amelia?"
+
+**Time Beetle**: "Calling tool: mission-amelia. Tool initiated. Amelia Earhart is a pilot and adventurer. She's known for her record-breaking flights and her disappearance in 1937. She was last seen flying over the Pacific Ocean. She's currently flying around the world in her plane, the Electra. She's running out of fuel and needs to find a place to land."
+
+**You**: "Device, can you help me find a place for Amelia to land?"
+
+**Time Beetle**: "Calling tool: find-landing-spot. Tool initiated. Searching for a suitable landing spot for Amelia Earhart. Please wait. Found a suitable landing spot. Coordinates: 7.5°N, 134.5°E. Amelia, I have found a suitable landing spot for you. Please head to the coordinates 7.5°N, 134.5°E."
+
+**Amelia Earhart**: "Thank you! I wish my device had that feature. I'll head there now."
+
+## Interact with Amelia Earhart
+
+If you want to interact with Ada, run the [Characters](/app/README.md) app.
+
+> [!IMPORTANT]
+> This is entirely fictional; the responses are generated by AI.
+> [Responsible AI disclaimer](/README.md#responsible-ai-disclaimer)
+
+<div>
+  <img src="./assets/amelia-front.jpeg" alt="Ada Lovelace" width="300">
+</div>
+
+**Steps**:
+
+1. Start a [![GitHub Codespace](https://img.shields.io/badge/GitHub-Codespace-brightgreen)](https://codespaces.new/microsoft/generative-ai-with-javascript)
+2. Navigate to _/app_ in the repo root.
+3. Locate the console and run `npm install` followed by `npm start`.
+4. Once it appears, select the "Open in Browser" button.
+5. Chat with Amelia.
+
+For a more detailed explanation of the app, see [Detailed app explanation](/lessons/01-intro-to-genai/README.md#interact-with-dinocrates).
+
+> [!NOTE]
+ > If you're running the project locally on your machine, please review the QuickStart guide to get a [GitHub personal access](/docs/setup/README.md#creating-a-personal-access-token-pat-for-github-model-access) token setup and replace the key in the code.
+
+## Tool calling
+
+**You**: "Device, what just happened?"
+
+**Time beetle**: "You just called a tool. A tool is a function that can be called by the AI model to perform a specific task. The tool can be used to perform a wide range of tasks, from simple calculations to complex operations. In this case, you called the `find-landing-spot` tool to help Amelia Earhart find a suitable landing spot."
```

Comment:
To make it consistent with the other chapters, let's change all occurences of "Time beetle" to "Time Beetle (uppercase "B").